notebook/notebookapp.py

```diff
@@ -1371,6 +1384,23 @@ def init_logging(self):
         logger.parent = self.log
         logger.setLevel(self.log.level)
     
+    def init_resources(self):
+        """initialize system resources"""
+        if resource is None:
+            self.log.debug('Ignoring min_open_files_limit because the limit cannot be adjusted (for example, on Windows)')
+            return
+
+        old_soft, old_hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+        soft = self.min_open_files_limit
+        hard = old_hard
+        if old_soft < soft:
+            if hard < soft:
+                hard = soft
+            self.log.debug(
+                'Raising open file limit: soft {}->{}; hard {}->{}'.format(old_soft, soft, old_hard, hard)
+            )
+            resource.setrlimit(resource.RLIMIT_NOFILE, (soft, hard))
```

Comment:
According to the documentation (https://docs.python.org/3/library/resource.html): 
> Raises ValueError if an invalid resource is specified, if the new soft limit exceeds the hard limit, or (*if a process tries to raise its hard limit*. ... A process with the effective UID of *super-user* can request any valid limit value

So it seems this will fail for non-root